### PR TITLE
fix(doc): Fix various doc warnings, part 2

### PR DESCRIPTION
**Pull request by upbqdn**
## Motivation

`cargo doc` currently produces a lot of warnings. We would like to incorporate cargo doc into our CI, so we should fix these warnings.

Previous PRs:
- #4514
- #4494

## Solution

I manually fixed various issues that were produced by `cargo doc --document-private-items`.

This PR doesn't fix all warnings yet because it would be too cumbersome to review it. There will be probably two more PRs that will deal with the remaining warnings.

### CODE REVIEW

File: zebra-network/src/isolated.rs [hunk 1]
```diff
@@ -124,6 +124,9 @@ where
 /// Transactions sent over this connection can be linked to the sending and receiving IP address
 /// by passive internet observers.
 ///
+/// TODO: Refactor [`connect_isolated_run_tor`](tor::connect_isolated_run_tor)
+/// below since the reference doesn't exist.
+///
 /// Prefer [`connect_isolated_run_tor`](tor::connect_isolated_run_tor) if available.
```
Comment (teor2345) [inline zebra-network/src/isolated.rs]: The function was renamed:

```suggestion
/// Prefer [`connect_isolated_tor`](tor::connect_isolated_tor) if available.
```

File: zebra-state/src/service.rs [hunk 2]
```diff
@@ -317,11 +321,16 @@ impl StateService {
         rsp_rx
     }
 
-    /// Update the [`LatestChainTip`], [`ChainTipChange`], and [`LatestChain`] channels
-    /// with the latest non-finalized [`ChainTipBlock`] and [`Chain`].
+    /// TODO: Update the reference to [`LatestChain`] below.
     ///
```
Comment (teor2345) [inline zebra-state/src/service.rs]: There's no type with this name, we never did that refactor.

```suggestion
```

File: zebra-state/src/service.rs [hunk 3]
```diff
@@ -317,11 +321,16 @@ impl StateService {
...
-    /// Returns the latest non-finalized chain tip height,
-    /// or `None` if the non-finalized state is empty.
+    /// Update the [`LatestChainTip`], [`ChainTipChange`], and [`LatestChain`]
```
Comment (teor2345) [inline zebra-state/src/service.rs]:
```suggestion
    /// Update the [`LatestChainTip`], [`ChainTipChange`], and `best_chain_sender`
```

File: zebra-state/src/service/non_finalized_state/chain/index.rs [hunk 4]
```diff
@@ -21,15 +21,18 @@ use super::{RevertPosition, UpdateWith};
 pub struct TransparentTransfers {
     /// The partial chain balance for a transparent address.
     ///
```
Comment (teor2345) [inline zebra-state/src/service/non_finalized_state/chain/index.rs]: This has already been done

```suggestion
```

File: zebra-state/src/service/non_finalized_state/chain/index.rs [hunk 5]
```diff
@@ -21,15 +21,18 @@ use super::{RevertPosition, UpdateWith};
...
-    /// TODO:
-    /// - to avoid [`ReadStateService`] response inconsistencies when a block has just been finalized,
-    ///   revert UTXO receives and spends that are at a height less than or equal to the finalized tip.
+    /// TODO: To avoid [`ReadStateService`](crate::service::ReadStateService)
+    /// response inconsistencies when a block has just been finalized, revert
+    /// UTXO receives and spends that are at a height less than or equal to the
+    /// finalized tip.
```
Comment (teor2345) [inline zebra-state/src/service/non_finalized_state/chain/index.rs]:
```suggestion
```

File: zebra-state/src/service/non_finalized_state/chain/index.rs [hunk 6]
```diff
@@ -39,9 +42,10 @@ pub struct TransparentTransfers {
...
     ///
-    /// TODO:
-    /// - to avoid [`ReadStateService`] response inconsistencies when a block has just been finalized,
-    ///   combine the created UTXOs, combine the spent UTXOs, and then remove spent from created
+    /// TODO: To avoid [`ReadStateService`](crate::service::ReadStateService)
+    /// response inconsistencies when a block has just been finalized, combine
+    /// the created UTXOs, combine the spent UTXOs, and then remove spent from
+    /// created
     ///
     /// Optional:
```
Comment (teor2345) [inline zebra-state/src/service/non_finalized_state/chain/index.rs]: This has already been done

```suggestion
    /// Optional TODOs:
```

**Review by teor2345 — COMMENTED** [review]
Looks good, just a few fixes we might want to apply.

Are we building docs with `--all-features`, or does that happen automatically already?

**Comment by upbqdn** [comment]
Thanks for resolving the TODOs I left in the PR. I added them because sometimes I wasn't sure how to fix certain warnings.

> Are we building docs with `--all-features`, or does that happen automatically already?

We're not doing that automatically. I forgot about this, so I was building the docs without that flag. I think we should add that to the CI. I double checked if using this flag adds new warnings, but thankfully it doesn't.

Now I'm building the docs with:
```bash
cargo d --document-private-items --all-features
```

**Comment by teor2345** [comment]
> > Are we building docs with `--all-features`, or does that happen automatically already?
> 
> We're not doing that automatically. I forgot about this, so I was building the docs without that flag. I think we should add that to the CI.

I added this change to ticket #4524.

**Review by teor2345 — APPROVED** [review]
Looks great!

**Comment by teor2345** [comment]
"Test all" failed because "restart_stop_at_height" seemed to hang without any output.

Maybe it's a network issue, I restarted it to try again. If it still doesn't work, we could open a ticket, and merge this PR manually.

**Comment by upbqdn** [comment]
Only the `lwd-full-sync` test fails now.

**Comment by teor2345** [comment]
> Only the `lwd-full-sync` test fails now.

I'm still seeing "Test all" time out after 6 hours, unfortunately the marker is the same colour as "skipped".

Screen Shot 2022-06-08 at 09 01 48

I opened PR #4580 to see if I could fix it.

**Review by conradoplg — APPROVED** [review]

**Comment by upbqdn** [comment]
@Mergify update.

**Comment by upbqdn** [comment]
@mergify update.

**Pull request merged**